internal/context/context.go

```diff
@@ -0,0 +1,52 @@
+package context
+
+import (
+	"context"
+	"news/internal/config"
+	"sync"
+)
+
+type AppCtx struct {
+	ctx        context.Context
+	cancelFunc context.CancelFunc
+	config     *config.Config
+	wg         *sync.WaitGroup
+}
+
+func NewAppCtx() (*AppCtx, error) {
+	config, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	return &AppCtx{
+		ctx:        ctx,
+		cancelFunc: cancelFunc,
+		config:     config,
+		wg:         &sync.WaitGroup{},
+	}, nil
+}
```

Comment:
why do you need the whole standalone package to handle contexts? 🤔

Reply:
My suggestion is to leave AppCtx structure: 
1) It needs sync.WaitGroup for different processes as indicator that the processes are stopped. In my case it is goroutine that makes requests to external feeds periodically and save articles and its details in mongo-db; 
2) In addition to first point, cancel function is needed to sent signal to stop processes.

3) I have made config outside of appCtx structure but it can be also as a field of appCtx. The idea is little bit the same as I mentioned before:
We want to run another process, we need wg, cancel function for correct shutdown function (1 and 2 point) and config to defined certain configuration. For these targets it is convenient to pass single pointer appCtx instead of multiple parameters;